cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/StepExecutionData.java

```diff
@@ -240,5 +255,19 @@ private void removeStepSerializableSessionObjects(final RunEnvironment runEnv) {
         );
     }
 
+    private void handleWorkerGroup(WorkerGroupStatement workerGroup, Context flowContext, RunEnvironment runEnv,
+                                   ExecutionRuntimeServices execRuntimeServices) {
+        Value workerGroupValue;
+        String expression = workerGroup.getExpression();
+        if (workerGroup.getFunctionDependencies() == null && workerGroup.getSystemPropertyDependencies() == null) {
+            workerGroupValue = ValueFactory.create(expression);
+        }
+        else {
+            workerGroupValue = scriptEvaluator.evalExpr(expression, flowContext.getImmutableViewOfVariables(),
+                    runEnv.getSystemProperties(), workerGroup.getFunctionDependencies());
+        }
+        execRuntimeServices.setWorkerGroupName(workerGroupValue.toString());
```

Comment:
not a very nice approach (can this be sensitive?) you will end up with *****

Reply:
It can't be sensitive